flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java

```diff
@@ -306,94 +257,6 @@ private TestJobCancellationHandler() {
 		}
 	}
 
-	private class TestJobExecutionResultHandler
-		extends TestHandler<EmptyRequestBody, JobExecutionResultResponseBody, JobMessageParameters> {
-
-		private final Iterator<Object> jobExecutionResults;
-
-		private Object lastJobExecutionResult;
-
-		private TestJobExecutionResultHandler(
-				final Object... jobExecutionResults) {
-			super(JobExecutionResultHeaders.getInstance());
-			checkArgument(Arrays.stream(jobExecutionResults)
-				.allMatch(object -> object instanceof JobExecutionResultResponseBody
-					|| object instanceof RestHandlerException));
-			this.jobExecutionResults = Arrays.asList(jobExecutionResults).iterator();
-		}
-
-		@Override
-		protected CompletableFuture<JobExecutionResultResponseBody> handleRequest(
-				@Nonnull HandlerRequest<EmptyRequestBody, JobMessageParameters> request,
-				@Nonnull DispatcherGateway gateway) throws RestHandlerException {
-			if (jobExecutionResults.hasNext()) {
-				lastJobExecutionResult = jobExecutionResults.next();
-			}
-			checkState(lastJobExecutionResult != null);
-			if (lastJobExecutionResult instanceof JobExecutionResultResponseBody) {
-				return CompletableFuture.completedFuture((JobExecutionResultResponseBody) lastJobExecutionResult);
-			} else if (lastJobExecutionResult instanceof RestHandlerException) {
-				return FutureUtils.completedExceptionally((RestHandlerException) lastJobExecutionResult);
-			} else {
-				throw new AssertionError();
-			}
-		}
-	}
-
-	@Test
-	public void testSubmitJobAndWaitForExecutionResult() throws Exception {
```

Comment:
Why is this test being removed?

Reply:
This test tests `ClientUtils.submitJobAndWaitForResult` which has been moved to test scope in this pr. We don't test testing utils, do we?

Reply:
Well that's true, but these tests were written back when these utils were used for the actual job submission. So shouldn't we instead port them to the new submission API?